stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt

```diff
@@ -1,31 +1,94 @@
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.call.Call
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.livedata.controller.ChannelControllerImpl
+import io.getstream.chat.android.test.TestCoroutineExtension
+import io.getstream.chat.android.test.randomCID
+import io.getstream.chat.android.test.randomString
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.test.runBlockingTest
-import org.junit.Test
-import org.junit.runner.RunWith
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
 
-@RunWith(AndroidJUnit4::class)
-internal class MarkReadImplTest : BaseConnectedIntegrationTest() {
+@ExperimentalCoroutinesApi
+internal class MarkReadImplTest {
+
+    companion object {
+        @JvmField
+        @RegisterExtension
+        val testCoroutines = TestCoroutineExtension()
+    }
+
+    private lateinit var chatDomain: ChatDomainImpl
+    private lateinit var client: ChatClient
+
+    private lateinit var sut: MarkReadImpl
+
+    @BeforeEach
+    fun init() {
+        client = mock()
+        chatDomain = mock {
+            on { scope } doReturn testCoroutines.scope
+        }
+        sut = MarkReadImpl(chatDomain)
+    }
+
+    @Test
+    fun `Given empty cid Should throw exception`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            sut.invoke("")
+        }
+    }
 
     @Test
-    fun read() = testCoroutines.dispatcher.runBlockingTest {
-        val channelControllerImpl = chatDomainImpl.channel(data.channel1.cid)
-        channelControllerImpl.handleEvent(data.newMessageFromUser2)
-        advanceUntilIdle()
-        var unreadCount = channelControllerImpl.unreadCount.getOrAwaitValue()
-        Truth.assertThat(unreadCount).isEqualTo(1)
-
-        val result = chatDomain.useCases.markRead(data.channel1.cid).execute()
-        advanceUntilIdle()
-        assertSuccess(result)
-        val lastRead = channelControllerImpl.read.getOrAwaitValue()?.lastRead
-        Truth.assertThat(lastRead).isEqualTo(data.messageFromUser2.createdAt)
-        unreadCount = channelControllerImpl.unreadCount.getOrAwaitValue()
-        Truth.assertThat(unreadCount).isEqualTo(0)
-        Truth.assertThat(channelControllerImpl.toChannel().unreadCount).isEqualTo(0)
+    fun `Given invalid cid Should throw exception`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            sut.invoke("no-colon")
+        }
+    }
+
+    @Test
+    fun `Given valid cid Should invoke markRead() on ChannelController`() = runBlockingTest {
+        val channelController = mock<ChannelControllerImpl>()
```

Comment:
It's always good to split test setup, test action and test assert by empty lines. It let you distinguish these blocks briefly